From eaf900f854def9f33d6a1b1670443ed96a51c10c Mon Sep 17 00:00:00 2001
From: "Darryl L. Pierce" <dpierce@redhat.com>
Date: Wed, 21 Jan 2015 10:45:06 -0500
Subject: [PATCH 10/12] txshift: switch to uint for workers

workaround missing size_t implementation of
po::value_semantic* create_value(T& val, const std::string& arg)
as it would conflict with when size_t == unsigned int even when
it would help for size_t == unsigned long

fixes https://issues.apache.org/jira/browse/QPID-6135

Contributed by Dan Horaz <dan@danny.cz>
---
 qpid/cpp/src/tests/txshift.cpp | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/qpid/cpp/src/tests/txshift.cpp b/qpid/cpp/src/tests/txshift.cpp
index bf85bee..6ec28c7 100644
--- a/qpid/cpp/src/tests/txshift.cpp
+++ b/qpid/cpp/src/tests/txshift.cpp
@@ -40,7 +40,7 @@ namespace tests {
 struct Args : public qpid::TestOptions
 {
     std::string workQueue;
-    size_t workers;
+    uint workers;
 
     Args() : workQueue("txshift-control"), workers(1)
     {
@@ -178,7 +178,7 @@ int main(int argc, char** argv)
             worker.run();
         } else {
             boost::ptr_vector<Worker> workers;
-            for (size_t i = 0; i < opts.workers; i++) {
+            for (uint i = 0; i < opts.workers; i++) {
                 workers.push_back(new Worker(connection, opts.workQueue));
             }
             std::for_each(workers.begin(), workers.end(), boost::bind(&Worker::start, _1));
-- 
2.1.0

